### PR TITLE
Ambika fix add permissions checks to disable buttons in user management

### PR DESCRIPTION
**Pull request by kabraambika**
# Description
image
Fixes # (bug list priority low )

## Related PRS (if any):
To test this frontend PR you need to checkout the development in backend repository.

## Main changes explained:
- Modify ResetPasswordButton.jsx to disable the button when the `resetPassword` permission is not enabled for the user, and display a tooltip on the disabled button with the message: "You don't have permission to reset the password."
- Update UserSearchPanel.jsx to show a tooltip with the message: "You don't have permission to create a new user" on the disabled 'create new user' button when the user lacks the `postUserProfile` permission.
- Update UserTableData.jsx to add checks on the delete button when the `deleteUserProfile` permission is not enabled. Also, include checks for the final day, and for the pause/resume button when the `changeUserStatus` permission is not enabled. Tooltips should be displayed on these buttons as well.
- Update test cases in UserTableData.test.js and ResetPasswordButton.test.js accordingly.

## How to test:
1. check into current branch `git checkout ambika-add-permission-user-management`
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as a owner user
5. go to dashboard -> other links → Permissions management
6. Click on manage user permissions
7. Choose your volunteer user name and give permission 'see all users' and scroll at the bottom and click submit button
8. Logout and log as volunteer user name
9. go to dashboard→ other links → User management
10. verify create new user button, active/inactive cell, pause button, set final day button, delete button, reset password button are disabled
11. verify this new feature works in [dark mode](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI)

### Screenshots of after implementation
- Disable and Tooltip for create new user
 image

- Disable click on active/inactive active cell
 image

- Disable and Tooltip for Pause button
 image

- Disable and Tooltip for set final day button
 image

- Disable and Tooltip for delete button
 image

- Disable and Tooltip for reset password button
 image

### CODE REVIEW

**Review by Muzammil603 — APPROVED** [review]
Everything is functioning as expected
https://github.com/user-attachments/assets/8192034b-0a0b-4ab0-9be2-c7a023bed7ed

**Review by ashmita123 — DISMISSED** [review]
I've tested the changes, and everything works as expected. The permission checks are functioning correctly, buttons are disabled for users who don’t have the necessary permissions. Great job on this feature!


https://github.com/user-attachments/assets/e87f381f-724a-4e38-94f8-dfcf5c669fcb

**Review by ashayk9 — APPROVED** [review]
I’ve tested the changes, and everything is functioning as expected. 
The permission checks are working properly, and buttons are correctly disabled for users lacking the required permissions.

https://github.com/user-attachments/assets/b39a4d7b-e1ea-4d49-b3be-e5a22fd1136b

**Review by nevatiarishabh — APPROVED** [review]
I have tested the PR. Everything is functioning as expected. Tooltips are rendered and the buttons are disabled appropriately when the user does not have permission.

https://github.com/user-attachments/assets/5414f8f7-4395-4075-88f2-0474a7d6bf7f

File: src/components/UserManagement/UserTableData.jsx [hunk 1]
```diff
@@ -185,20 +229,36 @@ const UserTableData = React.memo(props => {
...
             <button
               type="button"
+              id={"btn-delete-"+props.user._id}
               className="btn btn-outline-danger btn-sm"
               onClick={e => {
                 props.onDeleteClick(props.user, 'archive');
               }}
               style={darkMode ? { boxShadow: "0 0 0 0", fontWeight: "bold" } : boxStyle}
-              disabled={props.auth?.user.userid === props.user._id}
+              disabled={props.auth?.user.userid === props.user._i || !canDeleteUsers}
```
[the remark this answers is not in the document]
Comment (kabraambika) [reply src/components/UserManagement/UserTableData.jsx]: Hello @nathanah , Thanks for the thorough review and giving suggestion. Yes, I accidentally removed the characters and now it's fixed in changeset 913cef261e1036105968f0650bfd5bcc727a8edb in this PR.

**Review by nishitag5 — APPROVED** [review]
The changes are working as expected.

https://github.com/user-attachments/assets/e5df8d0d-e569-47ea-9d82-2eaf47ad6d40

**Review by Logeshwari-Renu — APPROVED** [review]
Tested the PR and the User Permissions Changes are disabled for user only with "see all user" access. Great work!

**Comment by ptrpengdev** [comment]
LGTM 

image

**Review by ptrpengdev — APPROVED** [review]

**Review by nathanah — APPROVED** [review]

**Review by lcohen730 — APPROVED** [review]
All required buttons are disabled and can see all tooltips mentioned in light and dark modes


https://github.com/user-attachments/assets/d79d3b9c-881e-4359-a787-af151db0f949

**Review by sam19981 — APPROVED** [review]
Changes are working as expected

Screenshot 2024-09-21 at 9 50 53 PM

**Review by nikhilgiri0226 — APPROVED** [review]
image

https://github.com/user-attachments/assets/6b208093-3987-4aee-9905-8e2fcf830b82

verified that create new user button, active/inactive cell, pause button, set final day button, delete button, reset password button are disabled

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**